testing_guide.md

```diff
@@ -0,0 +1,691 @@
+
+If you want to use a binary, make sure to grab the correct one for your system. [NA]
+
+[source-code]: https://github.com/bitcoin/bitcoin/releases/tag/v24.0rc1
+
+### 2. Compile Release Candidate
+
+**If you grabbed a binary, skip this step**
+
+Before compiling, make sure that your system has all the right dependencies installed. As this guide utilizes the Bitcoin Core GUI, you must compile support for the GUI and have the `qt5` dependency already installed. Since descriptor wallets are now the default, it is recommended that you compile with sqlite, so make sure you have installed the `sqlite3` dependency.
+
+To ensure we build in sqlite and gui support, pass the following options when configuring:
+
+```shell
+./autogen.sh
+./configure --with-gui=yes --with-sqlite=yes --enable-suppress-external-warnings
+# nproc does not exist on macOS so we create an alias for it if necessary
+if ! command -v nproc &> /dev/null; then alias nproc="sysctl -n hw.physicalcpu"; fi
+make -j "$(($(nproc)+1))"
+```
+
+For more information on compiling from source, here are some guides to compile Bitcoin Core for [UNIX/Linux](https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md), [macOS](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md), [Windows](https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md), [FreeBSD](https://github.com/bitcoin/bitcoin/blob/master/doc/build-freebsd.md), [NetBSD](https://github.com/bitcoin/bitcoin/blob/master/doc/build-netbsd.md), and [OpenBSD](https://github.com/bitcoin/bitcoin/blob/master/doc/build-openbsd.md).
+
+### 3. Setting up command line environment
+
+If you plan to use the command line, below are a few environment variable to set.
+
+First, create a temporary data directory
+
+```shell
+export DATA_DIR=/tmp/24-rc-test
+mkdir $DATA_DIR
+```
+
+Next, specify the following paths. For source compiled, start from the root of your release candidate directory and run:
+
+```shell
+export BINARY_PATH=$(pwd)/src
+export QT_PATH=$(pwd)/src/qt
+```
+
+For the downloaded binary, start from the root of the downloaded release candidate (cd ~/bitcoin-24.0rc1, for example) and run:
+
+```shell
+export BINARY_PATH=$(pwd)/bin
+export QT_PATH=$BINARY_PATH
+```
+
+To avoid specifying the data directory (`-datadir=$DATA_DIR`) on each command, bellow are a few extra variables to set.
+
+> :information_source: **Note:** If you are testing on signet, you can also include the `-signet` flag
+
+```shell
+alias bitcoind="$(echo $BINARY_PATH)/bitcoind -datadir=$DATA_DIR"
+alias cli="$(echo $BINARY_PATH)/bitcoin-cli -datadir=$DATA_DIR"
+alias qt="$(echo $QT_PATH)/bitcoin-qt -datadir=$DATA_DIR"
+```
+
+The commands throughout the rest of the guide will look like:
+
+```shell
+cli [cli args]
+
+# for starting bitcoin-qt
+qt [cli args]
+```
+
+### 4. Reset testing environment
+
+Between sections in this guide, it's recommended to stop your node and wipe the data directory. You can use the commands provided below.
```

Comment:
good point!

Reply:
I was expecting that commands could be chained but in practise this creates some concurrency issues because $DATA_DIR is deleted before shutdown is completed.